### ISSUE TITLE
Text overlaps on mobile for some posts

### ISSUE THREAD

**Issue by Zhao-Andy** [thepracticaldev/dev.to#506]
@nickytonline commented on [Thu Jul 12 2018](https://github.com/thepracticaldev/dev.to_core/issues/577)

# _BUG_
### Current/Expected Behavior
Title text should not overlap with post description.

### Additional Info (steps to replicate, device/browser, helpful links)
Happens on mobile. I'm on an iPhone 6 SE on Chrome. It only occurs with longer titles.

### Screenshots
![e0e81145-58f6-4bbd-9ed6-7c30ab86e1fa](https://user-images.githubusercontent.com/833231/42666767-0e7e1010-8615-11e8-88c5-693449db5a2e.png)

---

@joshi-chinmay commented on [Tue Jul 17 2018](https://github.com/thepracticaldev/dev.to_core/issues/577#issuecomment-405486697)

One solution is to add the `nowrap` property to the title `<h4>...</h4>` -
```
{
 white-space: nowrap;
 overflow: hidden;
 text-overflow: ellipsis;
}
```

Also, I tried to move the `<h4>...</h4>` inside the `<div class="content">...</div>`.  Then added some height to `<h4>`. That didn't work.

Another solution would be to have calculated height for every `<div>...</div>`. Such as `.tags`, `.content`, etc.
But the first solution is looking good to me. By that other post-blocks are unaffected.
I can fix that.

---

@benhalpern commented on [Tue Jul 17 2018](https://github.com/thepracticaldev/dev.to_core/issues/577#issuecomment-405612743)

For context on this, the reason this even happens in the first place is because in order to make the whole div clickable but also have the sub-links clickable, we use some slightly complicated CSS which sort of limits us.

Alternate solutions to this are definitely welcome.

---

@nickytonline commented on [Tue Jul 17 2018](https://github.com/thepracticaldev/dev.to_core/issues/577#issuecomment-405652997)

Whoever takes this on, you can develop the fix using Storybook. Just run `npm run storybook` then navigate to http://localhost:6006. Storybook stories are found in the `__stories__` folder. As you edit your files and save them, Storybook will hot reload your changes.

For a quick rundown of Storybook, see [Getting Started with Storybook for React](https://dev.to/nickytonline/getting-started-with-react-storybook-9jh).

Obviously test it in the full blown app after 😜. The snapshot will need to be updated as well probably. Run `npm run test:watch`. You should get an error about a snapshot not being up to date. Press the letter `u` to update the snapshot. After that, the snapshot test should pass.

---

@nickytonline commented on [Sun Jul 22 2018](https://github.com/thepracticaldev/dev.to_core/issues/577#issuecomment-406920162)

It looks like this got corrected sometime after this bug was filed. Not sure who/when it was fixed though. It looks like the layout of the posts on mobile changed.

![image](https://user-images.githubusercontent.com/833231/43053876-fc7db39e-8dfc-11e8-9d76-58e4ff3866a4.png)

---

@joshi-chinmay commented on [Sun Jul 22 2018](https://github.com/thepracticaldev/dev.to_core/issues/577#issuecomment-406927040)

Wow. That's great news @nickytonline.
But, I don't see it on my device. Caching issue, maybe?
![devto](https://user-images.githubusercontent.com/16207311/43055434-1cce4f02-8e69-11e8-81e8-1b1be2c6aa85.png)

---

@nickytonline commented on [Sun Jul 22 2018](https://github.com/thepracticaldev/dev.to_core/issues/577#issuecomment-406931771)

I did an empty cache and hard reload in Chrome and I still see the fixed screenshot. If you clear your cache, do you still have the issue?

---

@joshi-chinmay commented on [Mon Jul 23 2018](https://github.com/thepracticaldev/dev.to_core/issues/577#issuecomment-406936636)

Yes, I cleared the cache. Also opened the website in incognito! But it's not working.
Can you check the style which is applied to the title? I can't figure out what's fixing the issue on your device.

**Comment by Zhao-Andy** [comment]
@joshi-chinmay Moved the issue :)

**Comment by Link2Twenty** [comment]
This issue is caused by the user's name being too long which leads to the line being wrapped. A simple solution would be to do the same to them as to the tags.

https://github.com/thepracticaldev/dev.to/blob/2022d45fb129265011a5e05a20cf42db6cf8b6e3/app/assets/stylesheets/articles.scss#L310-L320

In here we'd add 

```CSS
overflow: hidden;
white-space: nowrap;
text-overflow: ellipsis;
```

Which would lead to the [post](https://dev.to/t/innovation#article-link-38963) looking like so

![image](https://user-images.githubusercontent.com/3534427/48781516-138c8c00-ecd4-11e8-957a-d1185567820d.png)

This does mean we lose the post date and part of the user's name. I guess we have to work out if that is preferable to trying to make more space.

**Comment by Link2Twenty** [comment]
Turns out someone has already fixed this, #502, with the same solution as me. Though it has conflicts.

**Comment by maestromac** [comment]
Resolved with #502
